src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderForSinglePartRequest.cs

```diff
@@ -22,12 +28,15 @@ namespace Microsoft.Health.Dicom.Core.Features.Store.Entries
     public class DicomInstanceEntryReaderForSinglePartRequest : IDicomInstanceEntryReader
     {
         private readonly ISeekableStreamConverter _seekableStreamConverter;
+        private readonly StoreConfiguration _storeConfiguration;
 
-        public DicomInstanceEntryReaderForSinglePartRequest(ISeekableStreamConverter seekableStreamConverter)
+        public DicomInstanceEntryReaderForSinglePartRequest(ISeekableStreamConverter seekableStreamConverter, IOptions<StoreConfiguration> storeConfiguration)
         {
             EnsureArg.IsNotNull(seekableStreamConverter, nameof(seekableStreamConverter));
+            EnsureArg.IsNotNull(storeConfiguration, nameof(storeConfiguration));
```

Comment:
nit: I think we should check value too, like: `EnsureArg.IsNotNull(storeConfiguration?.Value, nameof(storeConfiguration));`